tests/jballerina-unit-test/src/test/resources/test-src/object/ObjectProject/src/ObjectEquivalencyTest/object-equivalency.bal

```diff
@@ -626,3 +626,55 @@ function testInherentTypeViolationWithNilType() {
     ObjectWithAnyTypeVariables o2 = o1;
     o2.x = (); // panic
 }
+
+type NonClientObject object {
+    public string name;
+    public string id = "";
+
+    function __init(string name) {
+        self.name = name;
+    }
+    public function send(string message) returns error? {
+    }
+    public function receive(string message) {
+    }
+};
+
+type ClientObjectWithoutRemoteMethod client object {
+    public string name;
+    public string id = "";
+
+    function __init(string name) {
+        self.name = name;
+    }
+    public function send(string message) returns error? {
+    }
+    public function receive(string message) {
+    }
+};
+
+function testObjectEqualityBetweenNonClientAndClientObject() {
```

Comment:
Shall we also add a negative runtime test?

How about something like

```ballerina
type Foo object {
    function foo() {

    }
};

type Bar client object {
    remote function foo() {

    }
};

type Baz record {|
    Foo f;
|};

public function main() {
    Baz b = {f: new};
    record {|
        object {}...;
    |} r = b;

    r["f"] = new Bar();
}
```

Reply:
Done